core/executor/src/allocator.rs

```diff
@@ -108,6 +112,12 @@ impl FreeingBumpHeapAllocator {
 
 		self.total_size = self.total_size + item_size + 8;
 		trace!(target: "wasm-heap", "Heap size is {} bytes after allocation", self.total_size);
+		if self.total_size > self.historic_max {
```

Comment:
I am not sure if I can follow the big-picture here but from the looks of it: you are calling telemetry whenever you are finding a new mex heap usage. depending the call-site of this `if self.total_size > self.historic_max {}`, how frequently will it be triggered? 

Alternatively, why not keep the condition, safe the max if we find a new one, and only send at the very end once? 

(again, I have no clue what/where this code is and am just reviewing based on seeing 10 lines so I might be totally out of cotext here 😀)